### PR TITLE
Feature/angular vars v12

### PR DESCRIPTION
**Pull request by pedrojrivera**
# Description

Describe what this change achieves

## Issues Resolved
Ability to upgrade to Angular 12 and theming still works properly.

Note:
As other developers incorporate into their own projects, some additional overrides may be needed since not every Material component was tested.

### CODE REVIEW

**Comment by johannesjo** [comment]
Thank you very much! I try to have a look tomorrow!

File: projects/material-css-vars/src/lib/material-css-vars.service.ts [hunk 1]
```diff
@@ -155,7 +155,8 @@ export class MaterialCssVarsService {
         color = MaterialCssVarsService.LIGHT_TEXT_VAR;
       }
       return {
-        val: `var(${color})`,
+        val: this._getCssVarValue(color),
+        // val: `var(${color})`,
```
Comment (johannesjo) [inline projects/material-css-vars/src/lib/material-css-vars.service.ts:159]:
```suggestion
```

File: projects/material-css-vars/src/lib/material-css-vars.service.ts [hunk 2]
```diff
@@ -269,10 +270,12 @@ export class MaterialCssVarsService {
     const updates = this._calculateContrastColorsForCurrentValues(palettePrefix)
       .map(({contrastColorVar, hue}) => {
         return {
-          val: `var(${contrastColorVar})`,
+          // val: `var(${contrastColorVar})`,
```
Comment (johannesjo) [inline projects/material-css-vars/src/lib/material-css-vars.service.ts:273]:
```suggestion
```

File: projects/material-css-vars/src/lib/material-css-vars.service.ts [hunk 3]
```diff
@@ -269,10 +270,12 @@ export class MaterialCssVarsService {
...
+          val: this._getCssVarValue(contrastColorVar),
           name: `${palettePrefix + MaterialCssVarsService.CONTRAST_PREFIX}${hue}`,
         };
       });
+      console.log(updates);
```
Comment (johannesjo) [inline projects/material-css-vars/src/lib/material-css-vars.service.ts:278]:
```suggestion
```

**Review by johannesjo — APPROVED** [review]

**Comment by johannesjo** [comment]
I am running into several errors when trying to serve the demo app:

```
$ng serve
⠏ Generating browser application bundles (phase: building)...Unknown error from PostCSS plugin. Your current PostCSS version is 8.3.5, but postcss-preset-env uses 7.0.36. Perhaps this is the source of the error below.
⠋ Generating browser application bundles (phase: building)...Unknown error from PostCSS plugin. Your current PostCSS version is 8.3.5, but postcss-preset-env uses 7.0.36. Perhaps this is the source of the error below.
✔ Browser application bundle generation complete.

Initial Chunk Files | Names         |      Size
vendor.js           | vendor        |   5.77 MB
main.js             | main          | 142.15 kB
polyfills.js        | polyfills     | 128.62 kB
runtime.js          | runtime       |   6.86 kB
styles.js           | styles        |   2.44 kB

                    | Initial Total |   6.05 MB

Build at: 2021-07-02T12:08:26.540Z - Hash: 37c6abcd54556bb7bc23 - Time: 9710ms

./src/styles.scss - Error: Module build failed (from ./node_modules/mini-css-extract-plugin/dist/loader.js):
ModuleBuildError: Module build failed (from ./node_modules/postcss-loader/dist/cjs.js):
BrowserslistError: Unknown version 81 of android
    at Function.select (/home/johannes/www/angular-material-css-vars/node_modules/browserslist/index.js:1132:17)
    at /home/johannes/www/angular-material-css-vars/node_modules/browserslist/index.js:336:33
    at Array.reduce (<anonymous>)
    at resolve (/home/johannes/www/angular-material-css-vars/node_modules/browserslist/index.js:318:18)
    at browserslist (/home/johannes/www/angular-material-css-vars/node_modules/browserslist/index.js:444:21)
    at Browsers.parse (/home/johannes/www/angular-material-css-vars/node_modules/autoprefixer/lib/browsers.js:63:12)
    at new Browsers (/home/johannes/www/angular-material-css-vars/node_modules/autoprefixer/lib/browsers.js:46:26)
    at loadPrefixes (/home/johannes/www/angular-material-css-vars/node_modules/autoprefixer/lib/autoprefixer.js:98:20)
    at plugin (/home/johannes/www/angular-material-css-vars/node_modules/autoprefixer/lib/autoprefixer.js:109:20)
    at /home/johannes/www/angular-material-css-vars/node_modules/postcss-preset-env/index.js:443:157
    at processResult (/home/johannes/www/angular-material-css-vars/node_modules/webpack/lib/NormalModule.js:701:19)
    at /home/johannes/www/angular-material-css-vars/node_modules/webpack/lib/NormalModule.js:807:5
    at /home/johannes/www/angular-material-css-vars/node_modules/loader-runner/lib/LoaderRunner.js:399:11
    at /home/johannes/www/angular-material-css-vars/node_modules/loader-runner/lib/LoaderRunner.js:251:18
    at context.callback (/home/johannes/www/angular-material-css-vars/node_modules/loader-runner/lib/LoaderRunner.js:124:13)
    at Object.loader (/home/johannes/www/angular-material-css-vars/node_modules/postcss-loader/dist/index.js:142:7)

./src/app/app.component.scss - Error: Module build failed (from ./node_modules/postcss-loader/dist/cjs.js):
BrowserslistError: Unknown version 81 of android
    at Function.select (/home/johannes/www/angular-material-css-vars/node_modules/browserslist/index.js:1132:17)
    at /home/johannes/www/angular-material-css-vars/node_modules/browserslist/index.js:336:33
    at Array.reduce (<anonymous>)
    at resolve (/home/johannes/www/angular-material-css-vars/node_modules/browserslist/index.js:318:18)
    at browserslist (/home/johannes/www/angular-material-css-vars/node_modules/browserslist/index.js:444:21)
    at Browsers.parse (/home/johannes/www/angular-material-css-vars/node_modules/autoprefixer/lib/browsers.js:63:12)
    at new Browsers (/home/johannes/www/angular-material-css-vars/node_modules/autoprefixer/lib/browsers.js:46:26)
    at loadPrefixes (/home/johannes/www/angular-material-css-vars/node_modules/autoprefixer/lib/autoprefixer.js:98:20)
    at plugin (/home/johannes/www/angular-material-css-vars/node_modules/autoprefixer/lib/autoprefixer.js:109:20)
    at /home/johannes/www/angular-material-css-vars/node_modules/postcss-preset-env/index.js:443:157



** Angular Live Development Server is listening on localhost:4200, open your browser on http://localhost:4200/ **

```

Did you encounter these too? I searched for a fix, but was unable to make it work so far.


Probably related:
https://github.com/angular/angular-cli/issues/20967

**Comment by pedrojrivera** [comment]
@johannesjo I am not having this issue. I have no errors. But I have seen this error before. Here is how I have resolved it.

In your terminal session (bash) run the following commands:

1. rm -rf node_modules
2. Delete package-lock.json
3. npm install
4. npm audit fix
5. npm dedup

Now you should be able to do an npm start with no issues.

**Comment by johannesjo** [comment]
Ah alright. Didn't catch that you introduced npm in favor of yarn. I am fine with that and now it works.

Thank you very much again for your work on this! 

I

File: projects/material-css-vars/src/lib/_variables.scss [hunk 4]
```diff
@@ -101,97 +104,6 @@ $mat-css-default-light-theme: (
   --palette-warn-contrast-A200: var(--light-warn-text),
   --palette-warn-contrast-A400: var(--light-warn-text),
   --palette-warn-contrast-A700: var(--light-warn-text),
-  // BACKGROUND
-  --palette-background-status-bar: _mat-css-hex-to-rgb(map_get($mat-grey, 300)),
-  --palette-background-app-bar: _mat-css-hex-to-rgb(map_get($mat-grey, 100)),
```
Comment (pham93) [inline projects/material-css-vars/src/lib/_variables.scss:106]: Why are all these removed?
setVariable(----palette-background-app-bar, 'blue') doesn't work anymore.

**Comment by astone2014** [comment]
Thank you for your great work @pedrojrivera . I'm sorry I wasn't able to contribute more.

I've tested these changes on my application and it works great.

Some changes/observations I made during the update:
- Change `mat-color($primary, 500, 0.4)` to `mat-css-color(500, 0.4, 'primary')`
  - Could not get transparent color from `mat-color`
- Change `rgba(map-get($background, background));` to `map.get($background, background);`
  -  Must add `@use 'sass:map';`
- No longer could do `@include _mat-toolbar-color($primary);`
- Added Angular Material to main stylesheet `@use '~@angular/material'as mat;`

**Pull request merged**